src/material.h

```diff
@@ -75,4 +77,13 @@ class material_type
         const mat_burn_data &burn_data( size_t intensity ) const;
 };
 
+namespace materials
```

Comment:
I agree with codemime here, static class methods calling a global instance of the class is just a bit more icky than global functions calling the global class instance.  Given the way it's done, the namespaced functions are the lesser evil.

What I'd *like* to move to is an explicit top-level object (owned by main or the main game loop) that owns all these little factories, then we can stop having bizarre crashes due to static initialization order, and have clear scopes for everything.  I don't see a reasonable way to avoid having entity construction rely on global-scope methods that call those factories, but one thing at a time.